File: sol/solution/0001-0100/0034. Find First and Last Position of Element in Sorted Array/Solution.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def search_range(nums, target)
  find_bound = ->(first) {
    l, r, ans = 0, nums.length - 1, -1
    while l <= r
      mid = (l + r) / 2
      if nums[mid] == target
        ans = mid
        if first
          r = mid - 1
        else
          l = mid + 1
        end
      elsif nums[mid] < target
        l = mid + 1
      else
        r = mid - 1
      end
    end
    ans
  }
  [find_bound.call(true), find_bound.call(false)]
end

File: sol/solution/0001-0100/0034. Find First and Last Position of Element in Sorted Array/Solution.php
class Solution {
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function searchRange($nums, $target) {
        $findBound = function($first) use ($nums, $target) {
            $l = 0; $r = count($nums) - 1; $ans = -1;
            while ($l <= $r) {
                $mid = intdiv($l + $r, 2);
                if ($nums[$mid] == $target) {
                    $ans = $mid;
                    if ($first) $r = $mid - 1;
                    else $l = $mid + 1;
                } elseif ($nums[$mid] < $target) $l = $mid + 1;
                else $r = $mid - 1;
            }
            return $ans;
        };
        return [$findBound(true), $findBound(false)];
    }
}